src/js/components/HealthChecksComponent.jsx

```diff
@@ -184,10 +199,39 @@ var HealthChecksComponent = React.createClass({
                 }
                 fieldId={`${fieldsetId}.${i}.portIndex`}
                 label="Port Index"
-                value={row.portIndex}>
+                value={row.portIndex || 0}>
                 <input ref={`portIndex${i}`} {...numberInputAttributes} />
               </FormGroupComponent>
             </div>
+            <div className={portNumberClassSet}>
+              <FormGroupComponent
+                errorMessage={
+                  getErrorMessage(`${fieldsetId}.${i}.port`)
+                }
+                fieldId={`${fieldsetId}.${i}.port`}
+                label="Port Number"
+                value={row.port || 0}>
```

Comment:
port or portIndex is removed completely in the model, if the counterpart is select.